high_card.rb

```diff
@@ -1,5 +1,71 @@
+require 'pry'
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+ranks = ['A', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'J', 'Q', 'K']
+suits = %w[hearts spades clubs diamonds]
 
+deck_of_cards = []
 players = []
+cards_in_play = []
+number_cards = []
+winners = []
+
+suits.each do |suit|
+  ranks.each do |rank|
+    deck_of_cards.push([rank, suit])
+  end
+end
+
+loop do
+  puts "#{players.length} players so far. Enter a player name, or type 'play':"
+  name = gets.chomp
+  break if name == 'play'
+  players << name
+end
+
+players.each_index do |i|
+  shuffled_deck = deck_of_cards.shuffle
+  cards_in_play << { player: players[i], suit: shuffled_deck[i - 1][1], rank: shuffled_deck[i - 1][0] }
```

Comment:
This line is a little confusing. It kinda works, but only kinda. You're looping on the `players` index, so `i` becomes 0 for the first loop, and then 1 the second time.

Each loop you shuffle the deck and then "copy" the card at `shuffled_deck[i - 1]`:
 - First player: `shuffled_deck[-1]`
 - Second player: `shuffled_deck[0]`
 - Third player: `shuffled_deck[1]`
 - etc.

You're choosing a card from the randomized deck at a position offset by the player's index. That "works", but it's weird since the `shuffled_deck` was already randomized with `.shuffle`.

More importantly you're only copying the card info from that deck position instead of removing it from the shuffled deck. Using the current line as-is, there exists the chance that you shuffle the deck twice such that the same card comes out to multiple players.

To avoid that, we can create our `shuffled_deck` once outside of the loop, and instead of copying at the index we'll `.pop` a card off of the `shuffled_deck`, ensuring that each player gets a unique card.